naming/src/main/java/org/wildfly/httpclient/naming/HttpNamingServerObjectResolver.java

```diff
@@ -37,10 +36,11 @@ public HttpNamingServerObjectResolver(HttpServerExchange exchange) {
     }
 
     private URIAffinity createLocalURIAffinity(HttpServerExchange exchange) throws URISyntaxException {
-        InetSocketAddress localAddress = (InetSocketAddress) exchange.getConnection().getLocalAddress();
         StringBuilder uriStringBuilder = new StringBuilder();
-        uriStringBuilder.append(exchange.getRequestScheme()).append("://")
-                .append(localAddress.getHostName()).append(":").append(localAddress.getPort())
+        uriStringBuilder
+                .append(exchange.getRequestScheme())
+                .append("://")
+                .append(exchange.getHostAndPort())
                 .append("/wildfly-services");
```

Comment:
Remember `/wildfly-services` is the default value if nothing is explicitly configured for the `http-invoker` in the Undertow subsystem. So hard coding this value seems wrong.